tests/tests/test_mtls.py

```diff
@@ -76,7 +76,9 @@ def make_script_artifact(artifact_name, device_type, output_path):
 
 
 class TestClientMTLSEnterprise:
-    def common_test_mtls_enterprise(self, setup_ent_mtls, algorithm=None):
+    def common_test_mtls_enterprise(
+        self, setup_ent_mtls, algorithm=None, use_hsm=False
+    ):
```

Comment:
The new added logic here is independent of what was before (IOW, we have `if ...:` with no `else`). What was your motivation for putting it here?

I suggest we leave this function as it was, and move all the hsm related part in a different helper function. If I haven't overlooked something, this should all be doable independently (before calling `common_test_mtls_enterprise`)

Reply:
> The new added logic here is independent of what was before (IOW, we have `if ...:` with no `else`). What was your motivation for putting it here?

1. the fact that once we get the key URI we put it in the config, and then start the client.
2. the minimal required changes. it only changed a couple of lines there. (the diff got confused a bit by me moving one line up.)

let me think how to address your comment. I will come up with something.

Reply:
@lluiscampos ok, this is the best I could do, I separated most of the hsm logic into separate function. I hope it is ok!

Reply:
I wanted to start the conversation, no need to force-push so quickly.

> 1. the fact that once we get the key URI we put it in the config, and then start the client.

I thought that it is not a big deal to write into the config file twice.

> 2. the minimal required changes. it only changed a couple of lines there. (the diff got confused a bit by me moving one line up.)

IMHO this is not a priority: we "pay" adding more complexity to the method.

Being said the above, it is okey leaving it as it is, this is a low priority conversation.